src/sst/core/statapi/statintensity.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2009-2020 NTESS. Under the terms
+// of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Copyright (c) 2009-2020, NTESS
+// All rights reserved.
+//
+// This file is part of the SST software package. For license
+// information, see the LICENSE file in the top level directory of the
+// distribution.
+
+#ifndef _H_SST_CORE_INTENSITY_STATISTIC_
+#define _H_SST_CORE_INTENSITY_STATISTIC_
+
+#include "sst/core/sst_types.h"
+#include "sst/core/warnmacros.h"
+
+#include "sst/core/statapi/statbase.h"
+#include "sst/core/statapi/statoutput.h"
+#include "sst/core/statapi/stat3dviz.h"
+
+namespace SST {
+namespace Statistics {
+
+
+/**
+ * @brief The intensity_event struct
+ * A intensity_event event contains the collected data through the IntensityStatistic
+ * time  = this is the time of the event.
+ * intensity = this is the value (a double) that is written
+ *         at a given timepoint. Depending on configuration,
+ *         either intensity or level could be written as the color
+*/
+struct intensity_event {
+    uint64_t time_; // progress time
+    double intensity_;
+    intensity_event(uint64_t t, double intensity) :
+    time_(t), intensity_(intensity)
+    {
+    }
+
+};
+
+struct sorted_intensity_event {
+  intensity_event ie_;
```

Comment:
We should add a few more comments on what the exact difference is here between these events. 

An `intensity_event` should only ever be pushed back on a vector or list. It is not meant to be comparable and used in a map or set.  It is also local to a single statistic.

A `sorted_intensity_event` can be sorted by time.  It also contains a unique stat ID to distinguish events from different statistic collectors.

Reply:
👍 64880f3ca5aa3483cd44f29cd367f862d0ac8f2c